modules/configuration/OverallStateCollectorConfig.py

```diff
@@ -0,0 +1,11 @@
+import os
+import logging
+
+logger = logging.getLogger('esxi')
+
+
+class OverallStateCollectorConfig:
+
+    def __init__(self):
+        self.vcenter_username = os.environ['VCENTER_USER']
+        self.vcenter_password = os.environ['VCENTER_PASSWORD']

```

Comment:
This is used in both of your Config classes. Move it to `__init__` of BaseCollector and from the collectors run a `super.__init__()` to get it read out. We've got to remove this duplication